src/ucs/datastruct/mpool.h

```diff
@@ -76,6 +76,7 @@ struct ucs_mpool_data {
     size_t                 elem_size;       /* Size of element in the chunk */
     size_t                 alignment;       /* Element alignment */
     size_t                 align_offset;    /* Offset to alignment point */
+    double                 grow_factor;     /* Grow factor for number of elements per chunk */
```

Comment:
Currently only `1` and `2` are used. Is there any real use case for another value?
Maybe have a flag like `exponential_growth` instead? It would simplify code logic and make behavior more predictable.
@yosefe - WDYT?

Reply:
@gleon99 @ybedbug IMO growth factor is more generic and does not complicate code (multiplication instead of a branch)